### PR TITLE
chore(api): add full Docker setup for server deployment

### PR DESCRIPTION
**Pull request by benoit-bremaud**
## Summary

- Add `packages/api/docker-compose.yml`: orchestrates the API container with a named SQLite volume, GHCR image reference, and `restart: unless-stopped` policy
- Add `packages/api/.env.docker.example`: production-oriented env template (`NODE_ENV=production`, Swagger disabled, absolute DB path inside container)
- Add `.github/workflows/docker-build.yml`: validates the Dockerfile on every PR (build only), builds and pushes to GHCR on merge to `main` (tags: `latest` + `sha-<short-sha>`)
- Update `Makefile`: add `docker-build`, `docker-up`, `docker-down`, `docker-logs` targets

## Checklist

- [ ] Happy path + sad path + edge cases covered
- [ ] `tsc --noEmit` passes
- [ ] Build passes
- [ ] Existing tests still green
- [ ] No `any`, no inline styles introduced

Closes #

### CODE REVIEW

**Comment by benoit-bremaud** [comment]
@vitalikevin @clemoune-tech @Moooniie — Pour info : la PR #927 finalise la dockerisation du backend API.

Concrètement, ça ajoute :
- Un fichier `docker-compose.yml` dans `packages/api/` pour démarrer l'API sur un serveur avec une seule commande
- Un template d'environnement `\.env\.docker\.example` adapté à la production
- Un workflow CI qui vérifie automatiquement que l'image Docker se construit à chaque PR, et la publie sur le registre GitHub (GHCR) à chaque merge sur `main`
- Quatre commandes `make docker-*` pour gérer le conteneur en local

Objectif : pouvoir déployer l'API sur un serveur avec `docker compose pull && docker compose up -d` après chaque merge.

**Comment by benoit-bremaud** [comment]
Closing in favour of a clean branch rebased on main — see PR #928.

File: packages/api/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#   nano .env   # set JWT_SECRET
+# ==============================================================================
+
+services:
+  api:
+    build: .
+    image: ghcr.io/benoit-bremaud/brasse-bouillon-api:latest
+    container_name: brasse-bouillon-api
+    ports:
+      - "${PORT:-3000}:3000"
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/api/docker-compose.yml:18]: **Should Have — fixed in PR #928 (commit `90db2d7`).**

Valid point. `${PORT:-3000}:3000` hardcodes the container-side port to 3000, so setting `PORT=8080` makes the app bind 8080 while Docker still forwards to 3000 → unreachable API.

Fixed by using `${PORT:-3000}:${PORT:-3000}` so both sides track the env var consistently. Applied in the replacement PR #928 (`chore/docker-api-full-v2`), which supersedes this one due to a branch divergence issue (see note below).

File: .release-please-manifest.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "packages/mobile-app": "0.1.0-alpha1",
+  "packages/api": "0.0.1",
+  "packages/website": "0.0.1",
+  "packages/beer-encyclopedia": "0.0.1"
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply .release-please-manifest.json:5]: **Disagree — not part of this PR's intended scope.**

The `.release-please-manifest.json` change appeared in this diff due to a branch divergence issue: this branch was created from commits that were later squash-merged into `main`, causing the diff to include ~83 unrelated commits. The file was not intentionally modified here.

This change does not appear in the replacement PR #928 (`chore/docker-api-full-v2`), which is rebased cleanly on current `main` and contains only the 3 Docker-related commits.

**Comment by benoit-bremaud** [comment]
**Note re: Copilot review — 300-file limit exceeded**

Copilot could not review this PR because it exceeds the 300-file limit. This happened because the branch was created from commits that were squash-merged into `main` later, causing the diff to include ~83 unrelated commits.

This PR is superseded by **PR #928** (`chore/docker-api-full-v2`), which contains only the 3 intended Docker commits rebased cleanly on current `main`. Please review PR #928 instead. This PR (#927) should be closed.
